server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -21,28 +22,24 @@ package object server {
     * @tparam C the request type of the resulting service
     * @tparam D the response type of the resulting service
     */
-  type Middleware[F[_], A, B, C, D] = Service[F, A, B] => Service[F, C, D]
+  type Middleware[F[_], A, B, C, D] = Kleisli[F, A, B] => Kleisli[F, C, D]
 
   object Middleware {
-    def apply[F[_], A, B, C, D](f: (C, Service[F, A, B]) => F[D]): Middleware[F, A, B, C, D] = {
-      service =>
-        Service.lift { req =>
-          f(req, service)
-        }
-    }
+    def apply[F[_], A, B, C, D](f: (C, Kleisli[F, A, B]) => F[D]): Middleware[F, A, B, C, D] =
```

Comment:
Does anybody even use these?  They're kind of Finagle-like.  I have to stare at it every time I remember it's here to understand how it works.

Reply:
@deprecated?

Reply:
Please, lets get rid of funky type wrappers unless they are neccesary. Feel Similarly towards EntityBody not that anyone is asking.